src/lib.rs

```diff
@@ -256,6 +257,98 @@ impl FailPoint {
     }
 }
 
+#[derive(Default)]
+struct FailPointRegistry {
+    // TODO: remove rwlock or store *mut FailPoint
+    registry: RwLock<HashMap<String, Arc<FailPoint>>>,
+}
+
+lazy_static! {
+    static ref REGISTRY: FailPointRegistry = FailPointRegistry::default();
+}
+
+pub fn setup() {
```

Comment:
should add comments for all public APIs.

Reply:
I will fix all docs in another pr when all things are settled down.